torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -54,6 +56,70 @@ def reset_table(self) -> None:
 kernel_side_table = KernelSideTable()
 
 
+###############################################################################
+# Mutation Tracker
+
+
+@dataclasses.dataclass
+class MutationInfo:
+    mutated: bool = False
+    used_in_unknown: bool = False
+
+
+# Super basic mutation tracking pass that tracks which inputs are used in stores
+# It bails if any of the inputs are used in non tl.load/tl.store positions.
+class MutationTracker(ast.NodeVisitor):
+    def __init__(self, infos) -> None:
+        super().__init__()
+        self.infos = infos
+        self.in_load = False
+        self.in_store = False
+
+    def visit_Name(self, node):
+        if node.id not in self.infos:
+            return
+        if self.in_load:
+            pass
+        elif self.in_store:
+            self.infos[node.id].mutated = True
+        else:
+            self.infos[node.id].used_in_unknown = True
+
+    def visit_Call(self, node):
+        if (
+            isinstance(node.func, ast.Attribute)
+            and isinstance(node.func.value, ast.Name)
+            and node.func.value.id == "tl"
+        ):
+            if node.func.attr == "store":
+                assert self.in_store is False
+                self.in_store = True
+                self.generic_visit(node)
+                self.in_store = False
+                return
+            if node.func.attr == "load":
+                assert self.in_load is False
```

Comment:
I think, in indirect indexing, we can run into nested `tl.load`s? This will break then.

Reply:
Can you give an example?

Reply:
This would require us to restore state if there are nested stuff

Reply:
For example, smth like (similar to what exists in many jagged tensor Triton kernels):

```
index = tl.load(index_ptr + offsets)
data = tl.load(data_ptr + index, mask)
```

Maybe we could use a counter instead of a boolean flag for tracking inside nested `loads`?